pkg/client/outofband/client.go

```diff
@@ -312,38 +313,45 @@ func didServiceBlockFunc(p Provider) func() (*did.Service, error) {
 	return func() (*did.Service, error) {
 		// TODO https://github.com/hyperledger/aries-framework-go/issues/623 'alias' should be passed as arg and persisted
 		//  with connection record
-		_, verKey, err := p.LegacyKMS().CreateKeySet()
+		verKeyID, _, err := p.KMS().Create(kms.ED25519Type)
 		if err != nil {
-			return nil, fmt.Errorf("failed CreateSigningKey: %w", err)
+			return nil, fmt.Errorf("didServiceBlockFunc: failed to create SigningKey handle: %w", err)
 		}
 
 		s, err := p.Service(mediator.Coordination)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("didServiceBlockFunc: failed Coordinate Mediate service: %w", err)
 		}
 
 		routeSvc, ok := s.(mediator.ProtocolService)
 		if !ok {
-			return nil, errors.New("cast service to Route Service failed")
+			return nil, errors.New("didServiceBlockFunc: cast service to Route Service failed")
 		}
 
+		verKey, err := p.KMS().ExportPubKeyBytes(verKeyID)
```

Comment:
future PR: it'd be really handy to have a `CreateAndExportPubKeyBytes()` function...

Reply:
It could be added if there's demand for it in the future.